src/pure/message.ml

```diff
@@ -198,15 +197,16 @@ let body message =
     in
     let body_promise = Stream.read_until_close stream in
     message.body <- Some body_promise;
-    body_promise
+    Eio.Promise.await_exn body_promise
 
 let set_body message body =
-  message.body <- Some (Lwt.return body);
+  message.body <- Some (Eio.Promise.create_resolved (Ok body));
   match message.kind with
   | Request -> message.server_stream <- Stream.string body
   | Response -> message.client_stream <- Stream.string body
 
 let set_content_length_headers message =
+  (* TODO Restore.
```

Comment:
Seems like we can use `Eio.Promise.peek` to do something similar to `Lwt.poll` here?